### PR TITLE
Lichess login

### LINKED ISSUE #57: Add Login with Lichess
**Is your feature request related to a problem? Please describe.** :bulb:
Lichess now offers [OAuth](https://lichess.org/api#section/Authentication), which means a "Login with Lichess" button is within reach. And although it sounds ironic, it would be the epitome of foss+chess enthusiasm to include such a button.

**Describe the solution you'd like** :sparkles:
Add the "Login with Lichess" button (under the google one). All sensitive variables will be stored in the .env file so that the maintainers can replicate [the app](https://lichess.org/account/oauth/app/create) in production.

**Describe alternatives you've considered** :star2:
Not including the bu.... gosh, I can't even fathom such a thing 😨 

**Additional context** :memo:
♟️ + ☮️  = ❤️ 
[gsSoC '21]

### PR DESCRIPTION
**Pull request by JadeMaveric**
Closes #57

* Uses `passport-lichess` stratergy manage the OAuth flow with Lichess
When the user clicks the "Login with Lichess" button
1. A new tabs (window) opens to API_URL/auth/lichess
2. This activates the passport-lichess stratergy that redirects to oauth.lichess.org
3. lichess redirects the new tab to API_URL/auth/lichess/callback
4. Here the lichess stratergy verifies that the user exists and add user info to the request header
5. The next callback function uses this info to generate the JWT and sends back an HTML page with a script
6. The script uses `window.opener.postMessage()` to send the JWT data back to the parent tab

TODO:
[ ] - Lint code to adhere to standards
[ ] - Make sure existing user/pass accounts can't login using Lichess without first connecting it in profile

### CODE REVIEW

**Comment by JadeMaveric** [comment]
@fave77 Which version of Node should I use?

**Comment by fave77** [comment]
> @fave77 Which version of Node should I use?

Version 12+ (preferably 14)

**Comment by JadeMaveric** [comment]
Manually connecting Lichess IDs from settings, won't be straightforward. Lichess doesn't allow multiple callback urls, nor query parameters in the callback url

File: server/src/controllers/user.controller.js [hunk 1]
```diff
@@ -248,10 +265,101 @@ const updatePassword = async (req, res) => {
...
 
+// Called when signing in with Lichess
+const lichessSignIn = async (accessToken, refreshToken, lichessProfile, done) => {
+  try {
+    let user = null;
+    let profile = null;
+    let error = null;
+
+    // Find User|Profile using LichessID
+    profile = await Profile.findOne({ lichess: lichessProfile.username });
```
Comment (JadeMaveric) [inline server/src/controllers/user.controller.js]: This needs to be `lichessProfile.id`

**Review by JadeMaveric — COMMENTED** [review]
Just need to fix one typo, and then it'll be ready to review

**Review by fave77 — APPROVED** [review]
@JadeMaveric Looks good to me 👍🏻

**Pull request merged**